backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/semantics/rdf/ColumnToRDF.java

```diff
@@ -0,0 +1,47 @@
+package org.molgenis.emx2.semantics.rdf;
+
+import static org.eclipse.rdf4j.model.util.Values.iri;
+import static org.molgenis.emx2.semantics.rdf.ColumnTypeToXSDDataType.columnTypeToXSD;
+import static org.molgenis.emx2.semantics.rdf.IRIParsingEncoding.encodedIRI;
+
+import org.eclipse.rdf4j.model.util.ModelBuilder;
+import org.eclipse.rdf4j.model.vocabulary.DC;
+import org.eclipse.rdf4j.model.vocabulary.OWL;
+import org.eclipse.rdf4j.model.vocabulary.RDF;
+import org.eclipse.rdf4j.model.vocabulary.RDFS;
+import org.molgenis.emx2.Column;
+import org.molgenis.emx2.Table;
+
+public class ColumnToRDF {
+
+  // todo: unit is missing (which would also be a sdmx-attribute:unitMeasure, typed as an
+  // qb:AttributeProperty)
+  public static void describeColumns(ModelBuilder builder, Table table, String schemaContext)
+      throws Exception {
+    String tableContext = schemaContext + "/" + table.getName();
+    for (Column c : table.getMetadata().getColumns()) {
+      String columnContext = tableContext + "/column/" + c.getName();
+      // SIO:000757 = database column
+      builder.add(columnContext, RDF.TYPE, iri("http://semanticscience.org/resource/SIO_000757"));
+      builder.add(columnContext, RDF.TYPE, iri("http://purl.org/linked-data/cube#MeasureProperty"));
+      if (c.isReference() || c.isOntology()) {
+        builder.add(columnContext, RDF.TYPE, OWL.OBJECTPROPERTY);
```

Comment:
Both `c.isReference()` and `c.isOntology()` and reference objects instead of atomic values so OWL.OBJECTPROPERTY applies. Perhaps a subclass to 'flavour' the OBJECTPROPERTY as being Ontology would be nice at some point? but it would not be part of any standard

Reply:
Both `c.isReference()` and `c.isOntology()` and reference objects instead of atomic values so OWL.OBJECTPROPERTY applies. Perhaps a subclass to 'flavour' the OBJECTPROPERTY as being Ontology would be nice at some point? but it would not be part of any standard